### DISCUSSION TITLE
Getting a warning when used Map based derivation

### DISCUSSION THREAD

**Discussion by MetaVII** [mobxjs/mobx#3255] [Q&A] [answered]
Greetings!

I have a store class, configured strictly, and class member function with argument to get values from Map based class field (TodoStore in the example).

I use this function in a component, and get the "Derivation is created/updated without reading any observable value. " (TodoList in example)

If I use alike function to get values from array based class field, it's all OK. (You can uncomment getToDoFromArray in example)

So, I would prefer to use strictly configured store and Map based class field, am I doing something wrong? Is it any right way to do so without getting that warning?

[example link](https://codesandbox.io/s/classic-mobx-stores-with-hooks-forked-b8ev5?file=/src/components/TodoList.js:282-298)

**Comment by urugator** [comment, marked as answer]
The warning means you either forget `observer` somewhere or you forget to make something `obervable`. In your case it's the latter - you're missing `todosMap: observable` in `makeObservable`

EDIT: consider using eslint-plugin-mobx

**Comment by urugator** [comment]
`[MobX] Since strict-mode is enabled, changing (observed) observable values without using an action is not allowed. Tried to modify: Todos@1.todosMap.keys()`
looks like a bug to me, but I will have to investigate

**Reply by urugator** [reply]
Fixed by https://github.com/mobxjs/mobx/pull/3257

**Reply by MetaVII** [reply]
Thank you.
